src/Features/CSharp/Portable/MakeMethodSynchronous/CSharpMakeMethodSynchronousCodeFixProvider.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.MakeMethodSynchronous;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.MakeMethodSynchronous
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    internal class CSharpMakeMethodSynchronousCodeFixProvider : AbstractMakeMethodSynchronousCodeFixProvider
+    {
+        private const string CS1998 = nameof(CS1998); // This async method lacks 'await' operators and will run synchronously.
+
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(CS1998);
+
+        protected override bool IsMethodOrAnonymousFunction(SyntaxNode node)
+        {
+            return node.IsKind(SyntaxKind.MethodDeclaration) || node.IsAnyLambdaOrAnonymousMethod();
+        }
+
+        protected override SyntaxNode RemoveAsyncTokenAndFixReturnType(IMethodSymbol methodSymbolOpt, SyntaxNode node)
+        {
+            return node.TypeSwitch(
+                (MethodDeclarationSyntax method) => FixMethod(methodSymbolOpt, method),
+                (AnonymousMethodExpressionSyntax method) => FixAnonymousMethod(method),
+                (ParenthesizedLambdaExpressionSyntax lambda) => FixParenthesizedLambda(lambda),
+                (SimpleLambdaExpressionSyntax lambda) => FixSimpleLambda(lambda),
+                _ => node);
+        }
+
+        private SyntaxNode FixMethod(IMethodSymbol methodSymbol, MethodDeclarationSyntax method)
+        {
+            var newReturnType = method.ReturnType;
+            if (methodSymbol.ReturnType.Name == "Task")
```

Comment:
Usually we do this by checking for `methodSymbol.ReturnType.Equals(compilation.GetTypeByMetadataName("System.Threading.Task"))`